internal/pkg/scaffold/olm-catalog/descriptor.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package catalog
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/printer"
+	"go/token"
+	"go/types"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/operator-framework/operator-sdk/internal/pkg/scaffold"
+
+	olmapiv1alpha1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type fieldVals struct {
+	name, typ, tag, comments string
+}
+
+// setCRDDescriptorsForGV parses document and type declaration comments on
+// CRD types to populate a csv's 'crds.owned[].{spec,status}Descriptors' for
+// a given group and version.
+func setCRDDescriptorsForGV(crdDesc *olmapiv1alpha1.CRDDescription, gv schema.GroupVersion) error {
+	fset := token.NewFileSet()
+	ff := func(info os.FileInfo) bool {
+		return strings.HasSuffix(info.Name(), "_types.go")
+	}
+	dir := filepath.Join(scaffold.ApisDir, strings.Split(gv.Group, ".")[0], gv.Version)
+	pkgs, err := parser.ParseDir(fset, dir, ff, parser.ParseComments)
+	if err != nil {
+		// Don't return in error as other CSV components can still be generated.
+		if os.IsNotExist(err) {
+			return nil
+		}
+		log.Fatal(err)
+	}
+
+	allTypes := make(map[string][]fieldVals)
+	for _, pkg := range pkgs {
+		ast.Inspect(pkg, func(n ast.Node) bool {
+			switch x := n.(type) {
+			case *ast.GenDecl:
+				for _, spec := range x.Specs {
+					if spec, ok := spec.(*ast.TypeSpec); ok {
+						specName := spec.Name.Name
+						if specName == crdDesc.Kind {
+							crdDesc.Description = processComments(x.Doc.Text())
+						}
+						vals, ok := allTypes[specName]
+						if !ok {
+							vals = make([]fieldVals, 0)
+						}
+						switch st := spec.Type.(type) {
+						case *ast.StructType:
+							for _, field := range st.Fields.List {
+								typ := processType(fset, field.Type)
+								comments := processComments(field.Doc.Text())
+								names := field.Names
+								if len(names) == 0 {
+									names = []*ast.Ident{{Name: typ}}
+								}
+								for _, name := range names {
```

Comment:
I'm not 100% sure but if you have inlined field names, there will be more than one `*ast.Ident{}` in `field.Names`. Ex.:

```go
type Foo struct {
    Bar, Baz int
}
```

Will give two names. That's what I'm guessing since I don't see any doc notes about it.